src/WebJobs.Script.WebHost/WebJobsServiceCollectionExtensions.cs

```diff
@@ -122,6 +124,8 @@ public static IServiceProvider AddWebJobsScriptHost(this IServiceCollection serv
             builder.Register<IFileSystem>(_ => FileUtility.Instance).SingleInstance();
             builder.RegisterType<VirtualFileSystem>();
             builder.RegisterType<VirtualFileSystemMiddleware>();
+            builder.RegisterType<EncyptedHostAssignmentContextReader>().As<IEncyptedHostAssignmentContextReader>().SingleInstance();
```

Comment:
As commented above, just registering the LinuxContainerInitializationHostService should be sufficient - I don't think we need these two registered types